filebeat/fileset/fileset_test.go

```diff
@@ -41,17 +42,24 @@ func makeTestInfo(version string) beat.Info {
 	}
 }
 
-func getModuleForTesting(t *testing.T, module, fileset string) *Fileset {
+func getModuleForTesting(t *testing.T, module, fileset string, filesetCfg *FilesetConfig) *Fileset {
```

Comment:
Why `filesetCfg *FilesetConfig` was introduced if it's always `nil` in all usages? Can't we just always pass `nil` to the `New` function?

Reply:
it's not always nil:
 - https://github.com/AndersonQ/beats/blob/dc6126016cfa68a6c8b6b86e1506b6b803d7477f/filebeat/fileset/fileset_test.go#L107
 - https://github.com/AndersonQ/beats/blob/dc6126016cfa68a6c8b6b86e1506b6b803d7477f/filebeat/fileset/fileset_test.go#L243

that's why it was introduced as a parameter. `getModuleForTesting` ensures the configuration files will have the correct permissions, thus I changed `TestEvaluateVarsNginxOverride` and `TestGetInputConfigNginxOverrides` to call it instead of calling `New` directly